### ISSUE TITLE
Console logging does not display at runtime

### ISSUE THREAD

**Issue by yanne** [robotframework/RIDE#1217]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1178) by frankbp on 20 Dec 2012*

Executing the RF cases with RIDE which includes keyword to logging into console in runtime. The RIDE only display the console output at the end of the case execution but not at runtime.
Executing the same with RF itself via command line does not have this issue.

RobotFramework: 2.7.5
Python: 2.7.3
RIDE: 0.55

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1178#c1) by @mkorpela on 27 Dec 2012*


I'm closing this as WontFix for the following reasons.

There is no guarantee that the write operation will be done before the stream has been flushed. You can use sys.__stdout__.flush() to force flushing the output stream.

import sys

sys.__stdout__.write('something\n')
sys.__stdout__.flush()

If you use the python standard logging mechanism then you can see the message in message log.

import logging

logging.info('something')

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1178#c2) by frankbp on 27 Dec 2012*


Hi,

What I need is to log to console which built-in library logging does not
support so that I use Robot logging API.
And with that the messages coming to console does not display in message
box in RIDE until case finished but with command line interface the console
message will display in time.

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1178#c3) by @pekkaklarck on 9 Jan 2013*


Seems like a valid issue for me.

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1178#c4) by @mkorpela on 17 Jan 2013*


The basic problem is the one that I defined earlier. I will try to check if there is a way to solve this problem on RIDE side.

you can test this with the following library &lt;writings.py&gt;:
from robot.api import logger
from time import sleep
import sys

def delayed():
    logger.console('this comes only after the test ends')
    sleep(2)

def instant():
    logger.console('this comes instantly')
    sys.__stdout__.flush()
    sleep(2)

and the following test suite:
*** Settings ***
Library  mylib.py

*** Test Cases ***
Case 1
  delayed

Case 2
  instant

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1178#c5) by @mkorpela on 17 Jan 2013*


robot.api.logger.console:

def console(msg, newline=True):
    """Writes the message to the console.

    If the `newline` argument is `True`, a newline character is automatically
    added to the message.
    """
    if newline:
        msg += '\n'
    sys.__stdout__.write(msg)

Thus this operation doesn't flush output. Have to discuss if this is a bug in Robot Framework.

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=1178#c6) by @mkorpela on 17 Jan 2013*


Changed this to the robot side. We have to wait for the next Robot release.

<a href="https://code.google.com/p/robotframework/issues/detail?id=1335" rel="nofollow">https://code.google.com/p/robotframework/issues/detail?id=1335</a>

Workaround is to explicitly call flush:
import sys
import robot.api.logger

logger.console('something')
sys.__stdout__.flush()

I'm now closing this yet again as WontFix as this can not be fixed in RIDE.
